### ISSUE TITLE
heffte build failure with +hip

### ISSUE THREAD

**Issue by balay** [xsdk-project/xsdk-issues#265]
This build is using:
```
    version("2.5.0", branch="master")
```

And hip is installed by spack [as a dependency]
```
CMake Warning:
  Ignoring empty string ("") provided on the command line.


-- The CXX compiler identification is Clang 17.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /scratch/svcpetsc/spack.x/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/hip-6.1.2-zpsxdbnmfv5b63mi245ddhaavepcx5bh/bin/hipcc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.34.1")
-- Found MPI_CXX: /scratch/svcpetsc/spack.x/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/mpich-4.2.2-3oh4jzric2cxnqfnpyjjve4dhomidkns/lib/libmpicxx.so (found version "4.1")
-- Found MPI: TRUE (found version "4.1")
CMake Error at /scratch/svcpetsc/spack.x/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/cmake-3.30.2-jikehauocy7aezat2osqgmbamo6szxf4/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
Call Stack (most recent call first):
  /scratch/svcpetsc/spack.x/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/cmake-3.30.2-jikehauocy7aezat2osqgmbamo6szxf4/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /scratch/svcpetsc/spack.x/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.4.0/cmake-3.30.2-jikehauocy7aezat2osqgmbamo6szxf4/share/cmake-3.30/Modules/FindOpenMP.cmake:600 (find_package_handle_standard_args)
  cmake/FindHeffteFFTW.cmake:88 (find_package)
  CMakeLists.txt:139 (find_package)


-- Configuring incomplete, errors occurred!
```

[spack-build-out.txt](https://github.com/user-attachments/files/16895553/spack-build-out.txt)

ref:
```
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack.x$ git branch
* (HEAD detached at origin/xsdk-1.1.0)
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack.x$ ./bin/spack install -j64 xsdk@1.1.0+rocm amdgpu_target=gfx90a
```

**Comment by balay** [comment]
@mkstoyanov  this issue persists (with 2.4.1)

[spack-build-out.txt](https://github.com/user-attachments/files/17756326/spack-build-out.txt)

Let me know if you have suggestions.

For now - I'll continue with 'heffte~hip' in 'xsdk+hip' build.

**Comment by mkstoyanov** [comment]
Sorry I missed that. The issue is that the FFTW is compiled with OpenMP which is supported by the default C++ compiler. On the other hand, using HIP actually switches the compiler to AMD-clang and the version included in spack does not support OpenMP. There is OpenMP version of AMD-clang in spack, but that will not work as it is OpenMP with device offloading as opposed to the CPU threading used by FFTW.

Long story short, when building heffte+rocm we should disable FFTW.

**Comment by balay** [comment]
> Long story short, when building heffte+rocm we should disable FFTW.

Thanks!

Its building now with https://gitlab.com/xsdk-project/spack-xsdk/-/commit/b63c82cca4d8ad23f192bd8fd526b1353c1ab41c

[I'm not sure if there is a better syntax here with "xsdk_depends_on(,cuda_var=,rocm_var=)' notation - cc: @balos1

**Comment by mkstoyanov** [comment]
I may have to add another option for FFTW variants, threads OpenMP and so on, but that is for the future.

Meanwhile, heffte cannot use +rocm and +fftw at the same time in spack.
